### DISCUSSION TITLE
How do I use NextJS getServerSideProps with react query useInfiniteQuery

### DISCUSSION THREAD

**Discussion by peteregbujie** [TanStack/query#2633] [Q&A]
I am trying to implement infinite scroll in NextJS. So I need to replace useQuery with useInfiniteQuery. I just cannot figure out how to  use getServerSideProps with useInfiniteQuery. Pls help.

**Comment by TkDodo** [comment]
Please post a codesandbox example showing what you’re struggling with, then I can take a look :)

**Reply by peteregbujie** [reply]
Hello thank you for your response. Someone on reddit suggested I use prefetchInfiniteQuery. I am working on it at the moment. Will share the codebase in the coming days.

**Reply by TkDodo** [reply]
I think that was me 👋😅

**Reply by peteregbujie** [reply]
😅😅😅😅😅😅. It was you. I am working on the implementation you suggested. Thank you so much.
